rest-client/src/main/java/org/sagebionetworks/bridge/rest/ErrorResponseInterceptor.java

```diff
@@ -72,41 +74,46 @@ private void throwExceptionOnErrorStatus(String url, int statusCode, JsonElement
                 message = "There has been an error on the server";
             }
         }
-
-        BridgeSDKException e;
-        if (statusCode == 401) {
-            e = new NotAuthenticatedException(message, url);
-        } else if (statusCode == 403) {
-            e = new UnauthorizedException(message, url);
-        } else if (statusCode == 404) {
-            e = new EntityNotFoundException(message, url);
-        } else if (statusCode == 410) {
-            e = new UnsupportedVersionException(message, url);
-        } else if (statusCode == 412) {
```

Comment:
So, after we check all the known business exceptions which should be JSON, we have a generic exception which preserves the status code and a message if we can parse it, this seems sufficient. I'll add tests of what happens in this case (e.g. that a 500 throws BridgeSDKException with the right status code).